<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Login</title>
</head>

<body>
    <form id="loginForm">
        <div class="form-group">
            <label for="email">Email</label>
            <input type="text" name="email" id="email" value="gunz@mail.com">
        </div>
        <div class="form-group">
            <label for="password">Password</label>
            <input type="password" name="password" id="password" value="123">
        </div>
        <div class="form-group">
            <button>Kirim</button>
        </div>
    </form>
    <a href="forgot.html">Lupa password?</a>
    <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
</body>

</html>

<script>
    const url = 'https://be-2.gunz.my.id/auth/login';
    const todoUrl = 'https://be-2.gunz.my.id/todo';
    const logoutUrl = 'https://be-2.gunz.my.id/auth/logout';

    document.getElementById('loginForm').addEventListener('submit', function (e) {
        e.preventDefault();

        try {
            axios.post(url, {
                email: document.getElementById('email').value,
                password: document.getElementById('password').value,
            })
            .then(res => {
                const token = res.data.token;
                console.log(token);
    
                axios.get(todoUrl,{
                    headers: {
                        Authorization: `Bearer ${token}`,
                    },
                })
                .then(res => {
                    console.log(res);
    
                    axios.post(logoutUrl,{},{
                        headers: {
                            Authorization: `Bearer ${token}`,
                            // "Access-Control-Allow-Origin": "*",
                        },
                    })
                    .then(res => {
                        console.log(res)
                    })
                    .catch(err => console.log(err));
                })
                .catch(err => console.log(err));
    
            })
            .catch(err => console.log(err));
        } catch (error) {
            console.log(error);
        }
    })
</script>